lib/order.rb

```diff
@@ -0,0 +1,101 @@
+require 'csv'
+require_relative 'customer'
+
+class Order
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products # provided a hash with key prod name, value price)
+    @customer = customer # pass in from customer class
+    @fulfillment_status = fulfillment_status
+
+    raise ArgumentError.new("Invalid fulfillment status") if [:pending, :paid, :processing, :shipped, :complete].include?(@fulfillment_status) != true
+  end
+
+  def total
+    if @products.empty?
+      return 0
+    else
+      total = (@products.values.sum * 1.075).round(2)
+    end
+    return total
+  end
+
+  def add_product(product_name, price)
+    raise ArgumentError.new("product already exists") if @products.key?(product_name)
+    # another option: raise ArgumentError.new("product already exists") if @products.any? {|product, price| product == product_name}
+    @products[product_name] = price
+
+    return @products
+  end
+
+  def self.split_product_csv_helper(csv_products_line)
+    all_items = []
+
+    csv_products_line.split(/;/).each do |string| # splits at semi-colon
+      single_product = string.split(/:/) # split at the colon:
+
+      all_items << {single_product[0] => single_product[1].to_f}
+                                                  #.push vs. <<, does push return a new array so maybe I should use <<?  Or it doesn't matter?
+      # now we have: [{food1: 4.5}, {food2: 2.5}, etc.]
+    end
+    all_products = all_items.inject{|memo, obj| memo.merge(obj)}
```

Comment:
You can build up the Hash instead of converting an array of hashes into a single hash:

```suggestion
    all_products = {}

    csv_products_line.split(/;/).each do |string| # splits at semi-colon
      single_product = string.split(/:/) # split at the colon:

      all_products[single_product[0]] = single_product[1].to_f
    end
```

Also `<<` is an alias for `push`.